sdks/java/io/cdap/src/main/java/org.apache.beam.sdk.io.cdap/PluginBuilder.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.cdap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for building {@link Plugin} object.
+ */
+public class PluginBuilder {
+    private final Plugin plugin;
+
+    private final Class<?> pluginClass;
+    private Class<?> formatClass;
+    private Class<?> formatProviderClass;
+
+    private PluginConstants.PluginType pluginType;
+
+    private static final Logger LOG = LoggerFactory.getLogger(PluginBuilder.class);
+
+    /**
+     * Constructor for a plugin builder.
+     * @param pluginClass The main class of a plugin.
+     */
+    public PluginBuilder(Class<?> pluginClass) {
+        this.pluginClass = pluginClass;
+
+        try {
+            this.setPluginType();
+        } catch (IllegalArgumentException e) {
+            LOG.error("Error occurred while setting plugin class", e);
+        }
+
+        this.plugin = new Plugin();
```

Comment:
I thought we should create `Plugin` instance only in `build()` method after validation.